### PR TITLE
Stop the tooltip from covering the crosshair

### PR DESCRIPTION
**Pull request by envex**
## What does this implement/fix?

The `<LineChart />` tooltip would try and follow the cursor around when interacting with the chart. This allowed the tooltip to get into a state where it was covering the crosshair for the active state, which covers the data.

Now, we're going to make the tooltip never cover the crosshair. It will always try and render on either side of the crosshair that keeps the tooltip within the bounds of the chart.

## What do the changes look like?

**Before**

https://user-images.githubusercontent.com/149873/223225806-da4465fa-f095-49fd-b0f7-cf32c8474a95.mov

**After**

https://user-images.githubusercontent.com/149873/223225878-fe9d4823-375d-40e1-8cb8-516b7d0da4e1.mov

### Before merging

- [ ] Check your changes on a variety of [browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers) and devices.

- [x] Update the Changelog's Unreleased section with your changes.

- [x] Update relevant documentation, tests, and Storybook.

- [ ] Make sure you're exporting any new shared Components, Types and Utilities from the top level index file of the package

### CODE REVIEW

File: packages/polaris-viz/src/components/LineChart/utilities/getAlteredLineChartPosition.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  const newPosition = {...position};
+
+  let x = currentX;
+  let y = currentY;
+
+  //
+  // Y POSITIONING
+  //
+
+  if (props.isPerformanceImpacted) {
```
Comment (pbojinov) [inline packages/polaris-viz/src/components/LineChart/utilities/getAlteredLineChartPosition.ts:45]: Just curious, when does this prop become true?

**Review by pbojinov — APPROVED** [review]
Tophat and code looks good to me. Non-blocking but if you feel up to adding some tests on those util functions it could be use useful as a reference to the implementation or potential to catch any edge cases.

File: packages/polaris-viz/src/components/LineChart/utilities/getAlteredLineChartPosition.ts (hunk 1 above)
Comment (envex) [reply packages/polaris-viz/src/components/LineChart/utilities/getAlteredLineChartPosition.ts:45]: There's logic based on the amount of data points that will be rendered.

**Review by LBWright — APPROVED** [review]
